metal/resource_metal_connection.go

```diff
@@ -104,8 +104,8 @@ func resourceMetalConnection() *schema.Resource {
 			},
 			"project_id": {
 				Type:        schema.TypeString,
-				Required:    true,
-				Description: "ID of the project where the connection is scoped to, only used for type == \"shared\"",
+				Optional:    true,
+				Description: "ID of the project where the connection is scoped to. Required with type \"shared\"",
 				ForceNew:    true,
```

Comment:
Is this going to create an issue for users with dedicated connections that have a project id specified today? (Is that possible or was that only introduced within the alpha builds?)

Perhaps we should use custom validation to allow users to move from project to the same org without triggering a forcenew.  Or perhaps we should include a changelog note on how to convert to organization without losing the connection.

Reply:
It was introduces in alpha. Previously (3.2.2), "organization_id" was Required and "project_id" Optional and connection was always created through the org endpoint but validating (internally) that project was specified when type was ConnectionShared. Now, main difference is that if organization is not specified it will take the org to which the project belongs. So if anyone has created a dedicated connection with any alpha release (with org computed and with project required) now they can continue without org and with project because it first checks "organization_id" is set, otherwise it checks "project_id" is set to retrieve the organization id.

https://github.com/equinix/terraform-provider-metal/pull/223/files/207d44fa81df1ee0bb861dcdfd45e4527ada5914#diff-fb65e05d73d1a84ee2934a9c6033312470724448442603e716a9f6fada29c4eaR260-R266